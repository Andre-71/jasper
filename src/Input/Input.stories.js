import React from "react"
import { Formik, Form } from "formik"
import Input from "."

export default { title: "Input" }

const TestFormik = ({ children }) => (
  <Formik
    initialValues={{}}
    onSubmit={(values, actions) => {
      window.alert(JSON.stringify(values, null, 2))
      actions.setSubmitting(false)
    }}
  >
    {({ errors }) => (
      <Form>
        {children}
        {Object.keys(errors).map((key) => (
          <span key={key}>
            {key}: {errors[key]}
          </span>
        ))}
        <button type="submit">Submit</button>
      </Form>
    )}
  </Formik>
)

export const Text = () => (
  <TestFormik>
    <Input.Text name="TextInput" placeholder="Text Placeholder" />
  </TestFormik>
)

export const Radio = () => (
  <TestFormik>
    <Input.Radio name="RadioInput" value="one">
      1
    </Input.Radio>
    <Input.Radio name="RadioInput" value="two">
      2
    </Input.Radio>
  </TestFormik>
)

export const Number = () => (
  <TestFormik>
    <Input.Number name="NumberInput" placeholder="Number Placeholder" />
  </TestFormik>
)

export const TextArea = () => (
  <TestFormik>
    <Input.TextArea name="TextAreaInput" placeholder="TextArea Placeholder" />
  </TestFormik>
)

export const Phone = () => (
  <TestFormik>
    <Input.Phone name="PhoneInput" placeholder="Phone Placeholder" />
  </TestFormik>
)

export const Checkbox = () => (
  <TestFormik>
    <Input.Checkbox name="CheckboxInput" value="one">
      1
    </Input.Checkbox>
    <Input.Checkbox name="CheckboxInput" value="two">
      2
    </Input.Checkbox>
  </TestFormik>
)
