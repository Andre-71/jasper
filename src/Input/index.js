import React from "react"
import { Field } from "formik"
import {
  BaseInput,
  RadioButton,
  RadioSpan,
  PhoneSpan,
  CheckboxLabel
} from "./style"

export const TextInput = (props) => (
  <BaseInput type={props.type || "text"} {...props} />
)

export const RadioInput = ({ name, value, children, ...props }) => (
  <label style={{ cursor: "pointer" }}>
    <RadioButton
      type="radio"
      name={name}
      value={value || children}
      {...props}
    />
    <RadioSpan>{children}</RadioSpan>
  </label>
)

export const NumberInput = (props) => <BaseInput type="number" {...props} />

export const TextAreaInput = (props) => (
  <BaseInput component="textarea" {...props} />
)

export const PhoneInput = (props) => (
  <div style={{ position: "relative" }}>
    <PhoneSpan>+62</PhoneSpan>
    <BaseInput
      style={{
        position: "relative",
        zIndex: 10,
        paddingLeft: "4.5rem",
        width: "calc(100% - 19px - 4.5rem)"
      }}
      type="text"
      {...props}
      validate={(value) => {
        let errorMessage
        if (!value.match(/^\d*$/)) {
          errorMessage = "Invalid phone number"
        }
        return errorMessage
      }}
    />
  </div>
)

export const CheckboxInput = ({ value, children, ...props }) => (
  <CheckboxLabel>
    <Field type="checkbox" value={value || children} {...props} />
    <span className="checkmark" />
    {children}
  </CheckboxLabel>
)

export default {
  Checkbox: CheckboxInput,
  Number: NumberInput,
  Phone: PhoneInput,
  Radio: RadioInput,
  Text: TextInput,
  TextArea: TextAreaInput
}
