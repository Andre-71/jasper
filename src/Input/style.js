import styled from "styled-components"
import { Field } from "formik"

export const BaseInput = styled(Field)`
  width: calc(100% - 34px);
  padding: 10px 15px;
  color: #000;
  background-color: transparent;
  border: 2px solid #000;
  border-radius: 5px;
  font-family: inherit;
`

export const RadioSpan = styled.span`
  display: block;

  &:before {
    content: "";
    display: inline-block;
    width: 1.2rem;
    height: 1.2rem;
    vertical-align: -0.25rem;
    border-radius: 1.2rem;
    border: 1px solid #000;
    margin-right: 0.6rem;
    transition: background 0.2s ease;
    box-shadow: inset 0 0 0 0.15rem #fff;
  }
`

export const RadioButton = styled(Field)`
  border: 0;
  clip: rect(0 0 0 0);
  overflow: hidden;
  margin: 0;
  padding: 0;
  position: absolute;
  width: 1px;

  &:checked + ${RadioSpan}:before {
    background: #000;
  }
`

export const PhoneSpan = styled.span`
  position: absolute;
  margin: 15px;

  &:after {
    content: "";
    position: absolute;
    top: -5px;
    display: block;
    left: 2.5rem;
    width: 1px;
    height: calc(1rem + 10px);
    background: #000;
  }
`

