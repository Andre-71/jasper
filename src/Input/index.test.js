import React from "react"
import { shallow } from "enzyme"
import Input from "."

describe("Input.Text", () => {
  it("renders correctly", () => {
    const wrapper = shallow(
      <Input.Text name="input-name" placeholder="text placeholder" />
    )

    expect(wrapper).toMatchSnapshot()
  })
})

describe("Input.Radio", () => {
  it("renders correctly", () => {
    const wrapper = shallow(
      <Input.Radio name="input-name" value="input-value">
        Hello
      </Input.Radio>
    )

    expect(wrapper).toMatchSnapshot()
  })
})

describe("Input.Number", () => {
  it("renders correctly", () => {
    const wrapper = shallow(
      <Input.Number name="input-name" placeholder="number placeholder" />
    )

    expect(wrapper).toMatchSnapshot()
  })
})

describe("Input.TextArea", () => {
  it("renders correctly", () => {
    const wrapper = shallow(
      <Input.TextArea name="input-name" placeholder="placeholder" />
    )

    expect(wrapper).toMatchSnapshot()
  })
})

describe("Input.Phone", () => {
  it("renders correctly", () => {
    const wrapper = shallow(
      <Input.Phone name="input-name" placeholder="phone placeholder" />
    )

    expect(wrapper).toMatchSnapshot()
  })
})

describe("Input.Checkbox", () => {
  it("renders correctly", () => {
    const wrapper = shallow(
      <Input.Checkbox name="input-name">Hello</Input.Checkbox>
    )

    expect(wrapper).toMatchSnapshot()
  })
})

describe("Input.Dropdown", () => {
  const listDepartments = [
    "Finance",
    "Academy",
    "Competition",
    "Creative",
    "Event",
    "IT Dev",
    "Marketing",
    "Partnership",
    "Operational",
  ]

  it("renders correctly", () => {
    const wrapper = shallow(
      <Input.Dropdown
        placeholder={listDepartments[0]}
        choices={listDepartments}
        label="Departments"
      />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
