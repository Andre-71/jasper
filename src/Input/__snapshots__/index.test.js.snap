// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.Checkbox renders correctly 1`] = `
<styled.label>
  <Field
    name="input-name"
    type="checkbox"
    value="Hello"
  />
  <span
    className="checkmark"
  />
  Hello
</styled.label>
`;

exports[`Input.Dropdown renders correctly 1`] = `
<Field
  choices={
    Array [
      "Finance",
      "Academy",
      "Competition",
      "Creative",
      "Event",
      "IT Dev",
      "Marketing",
      "Partnership",
      "Operational",
    ]
  }
  component={[Function]}
  label="Departments"
  placeholder="Finance"
/>
`;

exports[`Input.Number renders correctly 1`] = `
<Styled(Field)
  name="input-name"
  placeholder="number placeholder"
  type="number"
/>
`;

exports[`Input.Phone renders correctly 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <styled.span>
    +62
  </styled.span>
  <Styled(Field)
    name="input-name"
    placeholder="phone placeholder"
    style={
      Object {
        "paddingLeft": "4.5rem",
        "position": "relative",
        "width": "calc(100% - 19px - 4.5rem)",
        "zIndex": 10,
      }
    }
    type="text"
    validate={[Function]}
  />
</div>
`;

exports[`Input.Radio renders correctly 1`] = `
<label
  style={
    Object {
      "cursor": "pointer",
    }
  }
>
  <Styled(Field)
    name="input-name"
    type="radio"
    value="input-value"
  />
  <styled.span>
    Hello
  </styled.span>
</label>
`;

exports[`Input.Text renders correctly 1`] = `
<Styled(Field)
  name="input-name"
  placeholder="text placeholder"
  type="text"
/>
`;

exports[`Input.TextArea renders correctly 1`] = `
<Styled(Field)
  component="textarea"
  name="input-name"
  placeholder="placeholder"
/>
`;
